File: frontend/src/models/gameAttempt.ts
import { TGameQuestion } from "./gameQuestion";
import { TUser } from "./user";

export type TGameAttempt = {
  attemptId: string;

  score: number;
  gameId: string;
  attemptedDate: string;
  attemptByUserId: string;
  attemptByUser: TUser;
  gameQuestions: TGameQuestion[];
};

export type TAddGameAttemptPayload = {
  gameId: string;
  attemptByUserId: string;
};

export type TCheckAnswerPayload = {
  questionId: string;
  answer: string;
};

File: backend/FinalAssignmentBE/Models/GameAttempt.cs
using System.ComponentModel.DataAnnotations;

namespace FinalAssignmentBE.Models;

public class GameAttempt
{
    [Key] public long AttemptId { get; set; }

    public int Score { get; set; } = 0;

    public DateTime AttemptedDate { get; set; } = DateTime.UtcNow;

    public long AttemptByUserId { get; set; }
    public User AttemptByUser { get; set; }
    public long GameId { get; set; }
    public Game Game { get; set; }

    public List<GameQuestion> GameQuestions { get; set; } = new List<GameQuestion>();

    public GameQuestion GenerateQuestion()
    {
        int min = 0, max = Game.NumberRange;
        Random random = new Random();
        var newQuestion = new GameQuestion(random.Next(min, max + 1), AttemptId);
        return newQuestion;
    }
}